sieve/bc_eval.c

```diff
@@ -1764,6 +1771,8 @@ int sieve_eval_bc(sieve_execute_t *exe, int is_incl, sieve_interp_t *i,
             if (res == SIEVE_RUN_ERROR)
                 *errmsg = "Fileinto can not be used with Reject";
 
+            implicit_keep &= cmd.u.f.copy;
```

Comment:
Bitwise logic is a bit confusing here, I'd do:

if (!cmd.u.f.copy) implicit_keep = 0;